_posts/2022-06-15-dirty-bytes-array-to-storage-bug.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+published: true
+title: 'Bug when Copying Dirty Bytes Arrays to Storage'
+date: '2022-06-15'
+author: Solidity Team
+category: Security Alerts
+---
+
+On July 1, 2021, a bug in the Solidity code generator was found by differential fuzzing.
+The bug causes the legacy code generation pipeline to generate code that
+may write dirty values to storage when copying ``bytes`` arrays from ``calldata`` or ``memory``.
+
+Initially, it was assumed that the dirty values in storage are only observable using inline
+assembly. However, resizing a ``bytes`` array using an empty ``.push()`` without actually
+writing values to it, can expose the dirty bytes without any use of inline assembly.
+
+The bug was already present in the initial implementation of ``bytes`` arrays in very early versions of the compiler.
+Despite this, it was never exploited or reported externally, and therefore we assigned it a severity of "low".
+
+## Which Contracts are Affected?
+
+Most instances of copying ``bytes`` arrays from ``memory`` or ``calldata`` to ``storage`` can be affected
+in the sense that they may write dirty values. Those dirty values in storage only become visible,
+if there are empty ``.push()``es to the ``bytes`` array (in older version also assignments to its
+``.length`` fields) and the resulting elements are assumed to be zero without actually writing to them.
```

Comment:
Not sure if https://github.com/ethereum/solidity-blog/pull/183#discussion_r896065153 will make it but if not, then we should correct this separately:
```suggestion
``.length`` field) and the resulting elements are assumed to be zero without actually writing to them.
```

Reply:
Adding this for now, then I'll consider the other comment.